scripts/sct_label_utils.py

```diff
@@ -850,93 +225,185 @@ def main(args=None):
+        # first pass use img as source
+        ref = Image(arguments.remove_reference)
+        out = sct_labels.remove_missing_labels(img, ref)
+
+        # second pass use previous pass result as reference
+        ref_out = sct_labels.remove_missing_labels(ref, out)
+        ref_out.save(path=ref.absolutepath)
     elif arguments.remove is not None:
-        process_type = 'remove'
-        value = arguments.remove
+        labels = arguments.remove
+        out = sct_labels.remove_labels_from_image(img, labels)
     elif arguments.keep is not None:
-        process_type = 'keep'
-        value = arguments.keep
-    else:
-        # no process chosen
-        sct.printv('ERROR: No process was chosen.', 1, 'error')
-    if arguments.msg is not None:
-        msg = arguments.msg+"\n"
-    else:
-        msg = ""
-    if arguments.o is not None:
-        input_fname_output = arguments.o
-    if arguments.ilabel is not None:
-        input_fname_previous = arguments.ilabel
+        labels = arguments.keep
+        out = sct_labels.remove_other_labels_from_image(img, labels)
+    elif arguments.create_viewer is not None:
+        msg = "" if arguments.msg is None else f"{arguments.msg}\n"
+        if arguments.ilabel is not None:
+            input_labels_img = Image(arguments.ilabel)
+            out = launch_manual_label_gui(img, input_labels_img, arguments.create_viewer, msg)
+        else:
+            out = launch_sagittal_viewer(img, arguments.create_viewer, msg)
+
+    out.save(path=output_fname, dtype=dtype)
+
+    if arguments.qc is not None:
+        generate_qc(fname_in1=input_filename, fname_seg=output_fname, args=args,
+                    path_qc=os.path.abspath(arguments.qc), dataset=arguments.qc_dataset,
+                    subject=arguments.qc_subject, process='sct_label_utils')
+
+
+def display_voxel(img: Image, verbose: int = 1) -> Sequence[Coordinate]:
+    """
+    Display all the labels that are contained in the input image.
+    :param img: source image
+    :param verbose: verbosity level
+    """
+
+    coordinates_input = img.getNonZeroCoordinates(sorting='value')
+    useful_notation = ''
+
+    for coord in coordinates_input:
+        sct.printv('Position=(' + str(coord.x) + ',' + str(coord.y) + ',' + str(coord.z) + ') -- Value= ' + str(coord.value), verbose=verbose)
+        if useful_notation:
+            useful_notation = useful_notation + ':'
+        useful_notation += str(coord)
+
+    sct.printv('All labels (useful syntax):', verbose=verbose)
+    sct.printv(useful_notation, verbose=verbose)
+
+    return coordinates_input
+
+
+# TODO [AJ] -> just print warnings or log or raise?
+def check_distance_between_labels(img: Image, max_dist_mm: float):
+    """
+    Calculate the distances between each label in the input image.
+    If a distance is larger than max_dist_mm, a warning message is displayed.
+    :param img: input image
+    :param max_dist: maximum distance (in mm) allowed between labels
+    """
```

Comment:
What was the rationale for keeping some functions (like this one) in `sct_label_utils` instead of moving them into `labels.py`?

Reply:
Initially there was a `ProcessLabels` class with a lot of labels processing logic. All of that was moved in the API module.

Then there were some free functions laying around in `sct_label_utils.py`. By default I kept all of them at the application level. I'll do another pass.